File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def yaksoo(number):
    results = []
    max_num = int(math.sqrt(number))
    for i in range(1, max_num + 1):
        if number % i == 0:
            results.append(i)
    return results
    

def solution(brown, yellow):
    # 노란격자 공약수 구하기
    results = yaksoo(yellow)
    
    for result in results:
        # 노란색 부분을 높이와 너비로 구하기
        height, width = result, yellow // result
        # 노란색 부분과 갈색 부분의 합이 맞을 때 break 하고 바로 내보내기
        if (width + 2) * (height + 2) - yellow == brown:
            answer = [width+2, height+2]
            break
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
// 약수 구하는 함수
function findGcd(num) {
  const arr = []
  for (let i = 1; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      arr.push(i)
    }
  }
  return arr
}

function solution(brown, yellow) {
  // 약수 구하기
  const gcd = findGcd(yellow)
  const result = [0, 0]
  for (i = 0; i < gcd.length; i++) {
    let l = gcd[i]
    let w = yellow / l
    let temp = 4 + (2 * l) + (2 * w)
    if (temp === brown) {
      result[0] = w + 2
      result[1] = l + 2
      break
    }
  }
  return result
}